File: convert-1d-array-into-2d-array/Solution.cpp
// https://leetcode.com/problems/convert-1d-array-into-2d-array

class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (n * m != original.size()) return vector<vector<int>>();
        vector<vector<int>> ret(m, vector<int> (n, 0));
        for(int i = 0; i < m; i++) {
            for(int j = 0; j < n; j++) {
                ret[i][j] = original[i * n + j];
            }
        }
        return ret;
    }
};

File: convert-1d-array-into-2d-array/Solution.py
// https://leetcode.com/problems/convert-1d-array-into-2d-array

class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if m * n != len(original):
            return []
        ret = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                ret[i][j] = original[i * n + j]
        return ret